src/main/java/com/minecolonies/coremod/client/gui/WindowHutUniversity.java

```diff
@@ -36,19 +44,105 @@ public WindowHutUniversity(final BuildingUniversity.View building)
-            button.setPosition(x + INITITAL_X_OFFSET, y + offset + INITITAL_Y_OFFSET);
-            view.addChild(button);
-            branches.add(branch);
+            List<IFormattableTextComponent> requirements = getHidingRequirementDesc(branch);
+            if(requirements.isEmpty())
+            {
+                final ButtonImage button = new ButtonImage();
+                button.setImage(new ResourceLocation(Constants.MOD_ID, MEDIUM_SIZED_BUTTON_RES));
+                button.setLabel(branch);
+                button.setSize(BUTTON_LENGTH, BUTTON_HEIGHT);
+                button.setTextColor(SLIGHTLY_BLUE);
+                button.setPosition(x + INITITAL_X_OFFSET, y + offset + INITITAL_Y_OFFSET);
+                view.addChild(button);
 
-            offset += button.getHeight() + BUTTON_PADDING;
+                offset += button.getHeight() + BUTTON_PADDING;
+            }
+            else
+            {
+                final Gradient gradient = new Gradient();
+                gradient.setSize(BUTTON_LENGTH, BUTTON_HEIGHT);
+                gradient.setPosition(x + INITITAL_X_OFFSET, y + offset + INITITAL_Y_OFFSET);
+                gradient.setGradientStart(239, 230, 215, 255);
+                gradient.setGradientEnd(239, 230, 215, 255);
+                gradient.setHoverToolTip(requirements);
+                view.addChild(gradient);
+                final ButtonImage button = new ButtonImage();
+                button.setImage(new ResourceLocation(Constants.MOD_ID, MEDIUM_SIZED_BUTTON_RES));
+                button.setLabel("-----------");
+                button.setSize(BUTTON_LENGTH, BUTTON_HEIGHT);
+                button.setTextColor(SLIGHTLY_BLUE);
+                button.setPosition(x + INITITAL_X_OFFSET, y + offset + INITITAL_Y_OFFSET);
+                view.addChild(button);
+
+                offset += button.getHeight() + BUTTON_PADDING;
+            }
+        }
+    }
+
+    /**
+     * Gets a list describing what requirements must be met to make at least one primary research for a branch visible.
+     *
+     * @param branch  The identifier for a branch.
+     * @return An empty list if at least one primary research is visible, or a list of IFormattableTextComponents describing the dependencies for each hidden primary research.
+     */
+    public List<IFormattableTextComponent> getHidingRequirementDesc(final String branch)
+    {
+        final List<IFormattableTextComponent> requirements = new ArrayList<>();
+        for(final String primary : IGlobalResearchTree.getInstance().getPrimaryResearch(branch))
+        {
+            if(!IGlobalResearchTree.getInstance().getResearch(branch, primary).isHidden()
+                 || IGlobalResearchTree.getInstance().isResearchRequirementsFulfilled(IGlobalResearchTree.getInstance().getResearch(branch, primary).getResearchRequirement(), building.getColony()))
+            {
+                requirements.clear();
+                break;
+            }
```

Comment:
return empty list here already?

Reply:
Yes. The goal is to have the branch locked until one Research is visible. If any one is complete, we don’t want to show a tooltip regarding the others, or require all of the primary research to be unlocked before you can start any of them.

Reply:
I mean, instead of break and clear, we can just return Collections.emptyList(), no?

Reply:
Oh. Yeah, derp on my part.